### PR TITLE
[Bullet] Mesh features

### PR DESCRIPTION
**Pull request by Lobotuerk**
Mesh collision seems to be working, at least with the duck.dae model.
ezgif-3-f39e68b5f641

### CODE REVIEW

File: bullet/src/ShapeFeatures.cc [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+                  vertices[indices[j+2]*3+2]);
+
+    mTriMesh->addTriangle(bv0, bv1, bv2);
+  }
+
+  btGImpactMeshShape *gimpactMeshShape =
+    new btGImpactMeshShape(mTriMesh);
+  gimpactMeshShape->updateBound();
+
+  // TODO(lobotuerk) Save collision if needed
```
Comment (Blast545) [inline bullet/src/ShapeFeatures.cc:75]: Isnt it neeeded to remove collision at the exit of the code?

File: bullet/src/ShapeFeatures.cc [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+  const auto &modelInfo = this->models.at(modelID);
+  const auto &link = linkInfo->link;
+  const auto &worldInfo = this->worlds.at(modelInfo->world);
+  const auto &world = worldInfo->world;
+  delete link->getCollisionShape();
+  link->setCollisionShape(gimpactMeshShape);
+
+  btGImpactCollisionAlgorithm::registerAlgorithm(worldInfo->dispatcher);
+
+  world->addRigidBody(link);
```
Comment (Blast545) [inline bullet/src/ShapeFeatures.cc:92]: Isn't it needed to remove the link once the collision is updated? I saw you removed its collision, I imagine you shouldn't need to add the link again (or if needed, remove it first)

File: bullet/src/ShapeFeatures.cc [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+  return this->AddCollision({_name, gimpactMeshShape, _linkID,
+                             modelID});
+
+}
+
+/////////////////////////////////////////////////
+Identity ShapeFeatures::CastToMeshShape(
+    const Identity &_shapeID) const
+{
+  ignerr << "CASTTOMESHSHAPE" << std::endl;
```
Comment (Blast545) [inline bullet/src/ShapeFeatures.cc:103]:
```suggestion
  ignwarn << "Dummy implementation CastToMeshShape" << std::endl;
```

**Review by Blast545 — APPROVED** [review]
Minor comments

File: bullet/src/ShapeFeatures.cc (hunk 1 above)
Comment (Lobotuerk) [reply bullet/src/ShapeFeatures.cc:75]: You can get the collision pointer from the rigid body

File: bullet/src/ShapeFeatures.cc (hunk 2 above)
Comment (Lobotuerk) [reply bullet/src/ShapeFeatures.cc:92]: Its only added here or in collision if the object has no mesh. So its only added on the last step

Comment (Blast545) [reply bullet/src/ShapeFeatures.cc:92]: This: `world->addRigidBody(link);` doesnt add a link already in the world for a second time?, not sure if I'm missing something

Comment (Lobotuerk) [reply bullet/src/ShapeFeatures.cc:92]: If the model has a mesh, it does not call sdfConstructCollision

**Pull request merged**